docs/mindsdb_sql/knowledge-bases.mdx

```diff
@@ -76,24 +168,154 @@ The overall `relevance_score` of a document is calculated as the sum of each chu
+### `storage`
 
-* `metadata_columns` provides the list of column names that will be used as the metadata of the knowledge base. Users can use these column names to insert them into the knowledge base and filter the search results by metadata. If not set, the metadata is not used.
+The vector store is a required component of the knowledge base. It stores data in the form of embeddings.
 
-* `content_columns` provides the list of column names that will be embedded as the content of the knowledge base. Users can use these column names to insert them into the knowledge base and filter the search results by content. If not set, the `content` column is used by default.
+It is optional for users to provide the `storage` parameter. If not provided, the default ChromaDB is created when creating a knowledge base.
 
-* `id_column` is the column name that will be used as the unique identifier by the knowledge base. If not set, the `id` column is used by default.
+In order to provide the storage vector database, it is required to connect it to MindsDB beforehand.
 
-<Info>
-When the `id_column` parameter is provided, the knowledge base uses this ID to identify rows from the source data.
+Here is an example for [PGVector](/integrations/vector-db-integrations/pgvector).
 
-When the `id_column` is not provided, the knowledge base uses an auto-generated sequence of numbers to identify rows from the source data. Note that these auto-generated IDs are duplicated in the following cases:
+```sql
+CREATE DATABASE my_pgvector
+WITH ENGINE = 'pgvector',
+PARAMETERS = {
+    "host": "127.0.0.1",
+    "port": 5432,
+    "database": "postgres",
+    "user": "user",
+    "password": "password",
+    "distance": "cosine"
+    };
 
-1. If the source data row is chunked into multiple chunks, then these rows in the knowledge base have the same auto-generated ID as they come from the same source data row.
-2. If the data from another data source is inserted, then the auto-generated sequence starts from zero again.
+CREATE KNOWLEDGE_BASE my_kb
+USING
+    ...
+    storage = my_pgvector.storage_table,
+    ...
+```
+
+Note that you do not need to have the `storage_table` created as it is created when creating a knowledge base.
+
+The available options include either [PGVector](/integrations/vector-db-integrations/pgvector) or [ChromaDB](/integrations/vector-db-integrations/chromadb).
+
+If the `storage` parameter is not provided, the system creates the default ChromaDB vector database called `<kb_name>_chromadb` with the default table called `default_collection` that stores the embedded data.
+
+### `metadata_columns`
+
+The data inserted into the knowledge base can be classified as metadata, which enables users to filter the search results using defined data fields.
```

Comment:
We may add more info as `The data inserted into the knowledge base which is not `content` can be classified as metadata,